src/core/compact_object.cc

```diff
@@ -108,8 +108,51 @@ size_t MallocUsedZSet(unsigned encoding, void* ptr) {
 }
 
 size_t MallocUsedStream(unsigned encoding, void* streamv) {
-  // stream* str_obj = (stream*)streamv;
-  return 0;  // TODO
+  auto* s = (stream*)streamv;
+  auto size = zmalloc_usable_size(s);
+  size += sizeof(*s);
+  size += s->rax_tree->numele * sizeof(streamID);
+  size += s->rax_tree->numnodes * sizeof(raxNode);
+
+  raxIterator ri;
+  raxStart(&ri, s->rax_tree);
+  raxSeek(&ri, "^", nullptr, 0);
+  size_t lpsize = 0;
+  while (raxNext(&ri)) {
+    auto* lp = (unsigned char*)ri.data;
+    lpsize += zmalloc_size(lp);
+  }
+  raxStop(&ri);
```

Comment:
I'm not entirely sure how streams work, but I guess this operation is not O(1), right?
All other types' memory usage is very quick, and I think(?) this is a requirement

Reply:
nope what you see here is not O(1) as it requires traversing the radix tree. Valkey approximates this with sampling. 

> All other types' memory usage is very quick, and I think(?) this is a requirement
Yes I agree for other types is more trivial but it's a data structure we don't control (we would need to change the radix tree to include how much memory it consumes and make this a constant operation). 

I am open to suggestions on how we can improve here

Reply:
First, you should verify if this is indeed a requirement. I _think_ it is, but it doesn't mean I'm right :)
Look in which flows we call this, and if it could be in hot paths etc.

If it is, then perhaps we should keep some cache for this value, and update it on the modification operations. We do something similar in Json, look for `JsonMemTracker`.

Reply:
it is a requirement. but this PR is not a high priority compared to other tasks (big value serialization)

Reply:
> If it is, then perhaps we should keep some cache for this value, and update it on the modification operations. We do something similar in Json, look for JsonMemTracker.

Yep +1

> it is a requirement. but this PR is not a high priority compared to other tasks (big value serialization)
+1